src/Microsoft.Restier.Core/Conventions/ConventionBasedApiModelBuilder.cs

```diff
@@ -95,7 +95,8 @@ private IQueryable GetEntitySetQuery(QueryExpressionContext context)
                 object target = null;
                 if (!entitySetProperty.GetMethod.IsStatic)
                 {
-                    target = context.QueryContext.GetApiService<ApiBase>();
+                    target = context.QueryContext.ApiContext.
```

Comment:
Conventions should be (optionally) applied to each Api type separately, so a conventional service instance should still handle a specific Api type.